bindings/python/src/_pyghex/py_dtype_to_cpp_name.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * ghex-org
+ *
+ * Copyright (c) 2014-2025, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <string>
+
+#include <pybind11/numpy.h>
+
+#include <gridtools/meta.hpp>
+#include <gridtools/common/for_each.hpp>
+
+#include <types.hpp>
+#include <util/demangle.hpp>
+
+namespace py = pybind11;
+
+namespace pyghex
+{
+
+std::string py_dtype_to_cpp_name(py::dtype dtype) {
+    std::string cpp_name;
+
+    gridtools::for_each<pyghex::types::data>([&cpp_name, &dtype](auto l) {
+        using type = decltype(l);
+
+        if (dtype.is(py::dtype::of<type>())) {
+            assert(cpp_name.empty());
+            cpp_name = util::demangle<type>();
+        }
+    });
+
+    if (cpp_name.empty()) {
+        throw std::invalid_argument("Unsupported numpy dtype");
+    }
+
```

Comment:
maybe better to handle the error at the call site (unless exceptions in C++ are commonly used in the bindings...)